packages/zoe/src/contracts/callSpread.js

```diff
@@ -0,0 +1,217 @@
+// @ts-check
+import '../../exported';
+
+import { assert, details } from '@agoric/assert';
+import { makePromiseKit } from '@agoric/promise-kit';
+import { E } from '@agoric/eventual-send';
+import {
+  assertProposalShape,
+  depositToSeat,
+  natSafeMath,
+  trade,
+  assertUsesNatMath,
+} from '../contractSupport';
+
+const { subtract, multiply, floorDivide } = natSafeMath;
+
+/**
+ * Constants for long and short positions.
+ *
+ * @type {{ LONG: 'long', SHORT: 'short' }}
+ */
+const Position = {
+  LONG: 'long',
+  SHORT: 'short',
+};
+
+const PERCENT_BASE = 100;
+const inverse = percent => subtract(PERCENT_BASE, percent);
+
+/**
+ * This contract implements a fully collateralized call spread. This is a
+ * combination of a call option bought at one strike price and a second call
+ * option sold at a higher price. The contracts are sold in pairs, and the
+ * purchaser pays the entire amount that will be paid out. The individual
+ * options are ERTP invitations that are suitable for resale.
+ *
+ * This option is settled financially. There is no requirement that the original
+ * purchaser have ownership of the underlying asset at the start, and the
+ * beneficiaries shouldn't expect to take delivery at closing.
+ *
+ * The issuerKeywordRecord specifies the issuers for four keywords: Underlying,
+ * Strike, Collateral and Options. The payout is in Collateral. Strike amounts
+ * are used for the price oracle's quotes as to the value of the Underlying, as
+ * well as the strike prices in the terms. Options indicates the
+ * invitationIssuer, which is part of the amounts of the options. The terms
+ * include { expiration, underlyingAmount, priceAuthority, strikePrice1,
+ * strikePrice2, settlementAmount }. expiration is a time recognized by the
+ * priceAuthority. underlyingAmount is passed to the priceAuthority, so it could
+ * be an NFT or a fungible amount. strikePrice2 must be greater than
+ * strikePrice1. settlementAmount uses Collateral.
+ *
+ * The creatorInvitation has terms that include the amounts of the two options
+ * as longOption and shortOption. When the creatorInvitation is exercised, the
+ * payout includes the two option positions, which are themselves invitations
+ * which can be exercised for free, and provide the option payouts.
+ *
+ * Future enhancements:
+ * + issue multiple option pairs with the same expiration from a single instance
+ * + create separate invitations to purchase the pieces of the option pair.
+ *   (This would remove the current requirement that an intermediary have the
+ *   total collateral available before the option descriptions have been
+ *   created.)
```

Comment:
There is currently no purchasing of the option pair in this contract. 

This goes back to my previous approach. I think it would still be valuable, as it's required in order to not have an intermediary have to front the funds.